packages/react/src/hooks/authentication.ts

```diff
@@ -1,83 +1,136 @@
-import { PasswordlessOptions, Provider } from '@nhost/client'
+import { useContext, useMemo } from 'react'
+
+import {
+  encodeQueryParameters,
+  PasswordlessOptions,
+  Provider,
+  ProviderOptions,
+  rewriteRedirectTo
+} from '@nhost/core'
 import { useSelector } from '@xstate/react'
 
-import { useNhost } from './common'
-import { useAuthenticated, useNhostInterpreter } from './common'
+import { NhostReactContext } from '../provider'
+
+import { useAuthenticated, useAuthInterpreter } from './common'
 
-export const useEmailPasswordSignIn = (stateEmail?: string, statePassword?: string) => {
-  const service = useNhostInterpreter()
-  const signIn = (valueEmail?: string | unknown, valuePassword?: string | unknown) =>
+export const useSignInEmailPassword = (
+  stateEmail?: string,
+  statePassword?: string,
+  stateOtp?: string
+) => {
+  const service = useAuthInterpreter()
+  const signInEmailPassword = (valueEmail?: string | unknown, valuePassword?: string | unknown) =>
```

Comment:
How about we use `email` and `password` instead of `valueEmail` and `valuePassword` to make it look better in an editor?

![image](https://user-images.githubusercontent.com/331818/160005992-59d7df13-4a1c-4522-a65c-e76d7f0ed932.png)

Reply:
In the `@nhost/nhost-js` SDK we always used objects as parameters. Maybe it's worth doing that here too?

```
signInEmailPassword({email, password});
```